src/index.js

```diff
@@ -22,17 +21,22 @@ class CameraRollRNPhotosFramework {
       eventEmitter.emit('onLibraryChange', changeDetails);
     });
 
+    //We need to make sure we clean cache in native before any calls
+    //go into RNPF. This is important when running in DEV because we reastart
+    //often in RN. (Live reload).
+    const methodsWithoutCacheCleanBlock = ['constructor', 'cleanCache', 'authorizationStatus', 'requestAuthorization'];
     const methodNames = (
       Object.getOwnPropertyNames(CameraRollRNPhotosFramework.prototype)
-        .filter(method => typeof method === 'function')
-        .filter(method => method !== 'constructor' && method !== 'cleanCache')
+        .filter(method => methodsWithoutCacheCleanBlock.indexOf(method) === -1)
     );
-
     methodNames.forEach(methodName => {
       const originalMethod = this[methodName];
       this[methodName] = function (...args) {
-        return cleanCachePromise.then(() => originalMethod.apply(this, args));
-      }
+        if(!this.cleanCachePromise) {
+          this.cleanCachePromise = RCTCameraRollRNPhotosFrameworkManager.cleanCache();
+        }
+        return this.cleanCachePromise.then(() => originalMethod.apply(this, args));
+      }.bind(this);
```

Comment:
no probs.. I seldom use plain old functions anymore.. to bad lambdas don't support ...args